posts/create-a-modern-web-app-using-vuejs-and-spring-boot-with-jhipster.md

```diff
@@ -0,0 +1,172 @@
+
+Here is a size comparison (minified) of the current popular frameworks ([more details here](https://gist.github.com/Restuta/cda69e50a853aa64912d)):
+
+```text
+| Name                     | Size  |
+| ------------------------ | ----- |
+| Angular 2                | 566K  |
+| Ember 2.2.0              | 435K  |
+| React 0.14.5 + React DOM | 133K  |
+| React 16.2.0 + React DOM |  97K  |
+| Vue 2.4.2                |  58K  |
+| Inferno 1.2.2            |  48K  |
+| Preact 7.2.0             |  16K  |
+```
+
+As I said before, the simplicity of Vue.js makes it very lightweight and easy to integrate with other libraries or existing projects. On the other hand, Vue.js is totally capable of building robust Single-Page Applications with the right tools and [JHipster](https://www.jhipster.tech/) is the perfect match for that. If you're not familiar with JHipster, I recommend watching [this video](https://www.youtube.com/watch?v=-VQ_SVkaXbs) that will give you a good introduction on what you can do with JHipster.
+
+# Vue.js using a JHipster blueprint
+
+By default, JHipster asks you to choose between Angular or React for the front-end framework. The concept of blueprint was introduced in version 5 and the goal is to extend the functionality of JHipster by letting you use your own sub-generator. More information on how to create and use a blueprint can be found in [the documentation](https://www.jhipster.tech/modules/creating-a-blueprint/).
+
+## Installation
+
+The official Vue.js blueprint will be used to generate the front-end. You can follow the installation instruction on [the blueprint repository](https://github.com/jhipster/jhipster-vuejs). Since the blueprint was still in development when I wrote this blog post, you may encounter some issues. Feel free to report them and submit a Pull Request if you think that a part was done the wrong way.
+
+## Application generation
+
+Let's start by creating a new folder and run JHipster with the vuejs blueprint:
+
+```bash
+mkdir vuejs-app
+cd vuejs-app
+jhipster --blueprint vuejs
+```
+
+The default answers can be selected for each questions and if the blueprint is correctly installed you should see this message:
+
+![](https://raw.githubusercontent.com/ippontech/blog-usa/master/images/2019/02/vuejs-jhipster.png)
+
+This [.yo-rc.json](https://raw.githubusercontent.com/Falydoor/blogpost-vuejs/master/.yo-rc.json) can be used in case you want to generate the exact same application as me. The file can be put in the application directory and then running `jhipster -d --blueprint vuejs` will generate the application without asking any questions.
```

Comment:
```suggestion
This [.yo-rc.json](https://raw.githubusercontent.com/Falydoor/blogpost-vuejs/master/.yo-rc.json) can be used in case you want to generate the same application referenced in this article. The file is placed in the application directory and then running `jhipster -d --blueprint vuejs` will generate the application without asking any questions.
```